phoenix-pherf/src/main/java/org/apache/phoenix/pherf/workload/mt/tenantoperation/TenantOperationFactory.java

```diff
@@ -0,0 +1,501 @@
+            try {
+                // Run the initialization operation.
+                OperationStats stats = impl.getMethod().apply(preRunSample);
+                LOGGER.info(phoenixUtil.getGSON().toJson(stats));
+            } catch (Exception e) {
+                LOGGER.error(
+                        String.format("Failed to initialize tenant. [%s, %s] ",
+                                tenantView.tenantId,
+                                tenantView.viewName
+                        ), e.fillInStackTrace());
+            }
+            tenantsLoaded.put(tenantView);
+        }
+
+        switch (input.getOperation().getType()) {
+        case NO_OP:
+            return new NoopTenantOperationImpl();
+        case SELECT:
+            return new QueryTenantOperationImpl();
+        case UPSERT:
+            return new UpsertTenantOperationImpl();
+        case USER_DEFINED:
+            return new UserDefinedOperationImpl();
+        default:
+            throw new IllegalArgumentException("Unknown operation type");
+        }
+    }
+
+    class QueryTenantOperationImpl implements TenantOperationImpl {
+
+        @Override public Function<TenantOperationInfo, OperationStats> getMethod() {
+            return new Function<TenantOperationInfo, OperationStats>() {
+
+                @Nullable @Override public OperationStats apply(@Nullable TenantOperationInfo input) {
+                    final QueryOperation operation = (QueryOperation) input.getOperation();
+                    final String tenantGroup = input.getTenantGroupId();
+                    final String opGroup = input.getOperationGroupId();
+                    final String tenantId = input.getTenantId();
+                    final String scenarioName = input.getScenarioName();
+                    final String tableName = input.getTableName();
+                    final Query query = operation.getQuery();
+                    final long opCounter = 1;
+
+                    String opName = String.format("%s:%s:%s:%s:%s", scenarioName, tableName,
+                            opGroup, tenantGroup, tenantId);
+                    LOGGER.info("\nExecuting query " + query.getStatement());
+                    // TODO add explain plan output to the stats.
+
+                    Connection conn = null;
+                    PreparedStatement statement = null;
+                    ResultSet rs = null;
+                    Long startTime = EnvironmentEdgeManager.currentTimeMillis();
+                    Long resultRowCount = 0L;
+                    Long queryElapsedTime = 0L;
+                    String queryIteration = opName + ":" + opCounter;
+                    try {
+                        conn = phoenixUtil.getConnection(tenantId);
+                        conn.setAutoCommit(true);
+                        // TODO dynamic statements
+                        //final String statementString = query.getDynamicStatement(rulesApplier, scenario);
```

Comment:
Can you explain what you mean by 'dynamic statements' and do we want to uncomment/remove this?